toolchest_client/tools/tool.py

```diff
@@ -222,21 +219,23 @@ def _preflight(self):
 
         # Check if the given output_path is a directory, if required by the tool
         # and if the user provides output_path.
-        if self._is_local_path():
+        if self._output_path_is_local():
             if self.output_is_directory:
                 if os.path.exists(self.output_path):
                     if os.path.isfile(self.output_path):
                         raise ValueError(
                             f"{self.output_path} is a file. Please pass a directory instead of an output file."
                         )
                 else:
-                    os.makedirs(self.output_path)
+                    os.makedirs(self.output_path, exist_ok=True)
+            else:
+                os.makedirs(os.path.dirname(self.output_path), exist_ok=True)
```

Comment:
Why are you making a dir if the output is not a directory here?

Reply:
Based on `os.path.dirname`, this command makes directories up until the file name (e.g., if the path is `/path/to/output/file.txt`, then `/path/to/output` gets created as a directory).

Reply:
Makes sense just skimmed a bit too much.